C/c4Index.cc

```diff
@@ -65,20 +65,39 @@ C4IndexUpdater::C4IndexUpdater(Retained<litecore::LazyIndexUpdate> u, C4Collecti
 
 C4IndexUpdater::~C4IndexUpdater() = default;
 
-size_t C4IndexUpdater::count() const { return _update->count(); }
+size_t C4IndexUpdater::count() const {
+    if ( hasFinished() ) {
+        WarnError("C4IndexUpdater::count() called on finished updater.");
+        return 0;
+    }
+    return _update->count();
+}
 
-FLValue C4IndexUpdater::valueAt(size_t i) const { return _update->valueAt(i); }
+FLValue C4IndexUpdater::valueAt(size_t i) const {
+    if ( hasFinished() ) {
+        litecore::error::_throw(litecore::error::NotOpen, "C4IndexUpdater::valueAt() called on finished updater.");
+    }
+    return _update->valueAt(i);
+}
 
 void C4IndexUpdater::setVectorAt(size_t i, const float* vector, size_t dimension) {
+    if ( hasFinished() ) {
+        litecore::error::_throw(litecore::error::NotOpen, "C4IndexUpdater::setVectorAt() called on finished updater.");
+    }
     _update->setVectorAt(i, vector, dimension);
 }
 
-void C4IndexUpdater::skipVectorAt(size_t i) { return _update->skipVectorAt(i); }
+void C4IndexUpdater::skipVectorAt(size_t i) {
+    if ( hasFinished() ) {
+        litecore::error::_throw(litecore::error::NotOpen, "C4IndexUpdater::SkipVectorAt() called on finished updater.");
```

Comment:
There is not much need to throw an exception here. Just log and return. An exception provides no value and wastes cycles with try / catch.

Reply:
same answer as above

Reply:
I didn't notice this returned a boolean but it is now a sort of useless one since the problem is not actually known anymore

Reply:
I have discussed with @borrrden before but we have thought that the behavior will be as follows from CBL API if the updater is finished:

count() : Return zero
valueAt() : Return null
setVector : No ops
skipVector : No ops
finish() : Throws CouchbaseLiteException with NotOpen error to indicate the mistake.

Reply:
As we do treat this an error (error also means an error needs to be returned), can we return true for no-ops?

Reply:
Yes, I don't think there is any issue with just returning and letting the method return true for this case.  So back to my original recommendation.